android/src/main/java/com/razorpay/flutter_customui/RazorpayDelegate.java

```diff
@@ -621,18 +652,122 @@ public void onSuccess(@NonNull List<UpiAccount> upiAccounts) {
             public void onError(@NonNull Error error) {
                 pendingResult.error(error.getErrorCode(), error.getErrorDescription(), toJsonString(error));
             }
-        },color);
+        }, color);
+    }
+
+    public void prefetchAndLinkNewUpiAccountUI(String customerMobile, String color, Result result, EventChannel.EventSink eventSink) {
+        try {
+            this.pendingResult = result;
+            this.eventSink = eventSink;
+            razorpay.upiTurbo
+                    .setCustomerMobile(customerMobile)
+                    .setColor(color)
+                    .prefetchAndLinkUpiAccountsWithUI(new UpiTurboPrefetchLinkAccountsResultListener() {
+                        @Override
+                        public void onResponse(AllAccounts allAccounts) {
+                            List<BankAccount> pinNotSetArr = new ArrayList<>();
+                            List<Object> pinSetArr = new ArrayList<>();
+
+                            if (allAccounts.getAccountsWithPinNotSet() != null) {
+                                for (BankAccount account : allAccounts.getAccountsWithPinNotSet()) {
+                                    pinNotSetArr.add(account);
+                                }
+                            }
+
+                            if (allAccounts.getAccountsWithPinSet() != null) {
+                                try {
+                                    pinSetArr.addAll(allAccounts.getAccountsWithPinSet());
+                                } catch (Exception e) {
+                                    Log.e("Exception Occurred", "Pin-set account parsing Exception");
+                                }
+                            }
+
+
+                            Map<String, Object> finalDict = new HashMap<>();
+                            finalDict.put("accountsWithPinNotSet", pinNotSetArr);
+                            finalDict.put("accountsWithPinSet", pinSetArr);
+
+                            String finalDictStr = new Gson().toJson(finalDict);
```

Comment:
@KausthubhRzr can you please rename `finalDictStr` variable as this data is not being sent as string to flutter wrapper.

Reply:
@ramprasadAnand in the above code snippet,the HashMap is getting converted to String. Hence I have added the name in the manner